user/utest/misbehave.c

```diff
@@ -24,3 +27,44 @@ int test_misbehave() {
 
   return 0;
 }
+
+static inline void exec_cp0_instr(void) {
+  int value;
+  asm volatile("mfc0 %0, $12, 0" : "=r"(value));
+}
+
+static inline void exec_reserved_instr(void) {
+  int value;
+  asm volatile("mfc2 %0, $12, 0" : "=r"(value));
+}
+
+static int spawn_process(void (*proc_handler)(void)) {
+  int pid;
+  switch ((pid = fork())) {
+    case -1:
+      exit(EXIT_FAILURE);
+    case 0:
+      proc_handler();
+      exit(EXIT_SUCCESS);
+    default:
+      return pid;
+  }
+}
+
+int test_userspace_cop_unusable(void) {
```

Comment:
Hence code for both functions is almost the same... please replace it with call to:
```c
static void check_exception(void (*code)(void), int signum)
```
... and move relevant code there.

Reply:
Sure, I'll fix it.